BonApp.js

```diff
@@ -47,306 +45,305 @@ render: function(output) {
+          sortStatus = 1
         }
 
-        return sortStatus;
-      };
-    };
+        return sortStatus
+      }
+    }
 
 
     var updateTheData = (function(_this) {
       return function(cafeId) {
-        var theCommand = 'curl -s \'http://legacy.cafebonappetit.com/api/2/menus?cafe=' + cafeId + "'";
+        var theCommand = 'curl -s \'http://legacy.cafebonappetit.com/api/2/menus?cafe=' + cafeId + "'"
 
         return _this.run(theCommand, function(err, myOut) {
-          makeItAllHappen(myOut);
-          return changeTheView();
-        });
-      };
-    })(this);
+          makeItAllHappen(myOut)
+          return changeTheView()
+        })
+      }
+    })(this)
 
 
     var initSearch = function() {
-      return jQuery.get('BonApp-widget/scripts/data.json', function(data) {
+      return jQuery.get('BonApp-Widget/scripts/data.json', function(data) {
         return $('#search').autocomplete({
           source: data.sort(sortByProperty('label')),
-          minLength: 3,
+          minLength: 2,
 
           select: function(event, ui) {
-            $("#spinner").show();
-            $('#search').val(ui.item.label);
+            $("#spinner").show()
+            $('#search').val(ui.item.label)
 
-            updateTheData(ui.item.id);
+            updateTheData(ui.item.id)
 
-            return false;
+            return false
           },
 
           open: function() {
             return $('.ui-autocomplete:visible').css({
               top: '+=0',
               left: '-=0'
-            });
+            })
           }
 
         }).autocomplete('instance')._renderItem = function(ul, item) {
-          return $('<li>').append('<a>' +
+          return $('<li>').append('<div>' +
                                   '<a id=\'search-label\'>' + item.label + '</a>' +
-                                  '<a id=\'search-description\'>' + item.desc + '</a>' + '</a>')
-                          .appendTo(ul);
-        };
-      });
-    };
+                                  '<a id=\'search-description\'>' + item.desc + '</a>' + '</div>')
+                          .appendTo(ul)
+        }
+      })
+    }
 
 
     var cleanUpStationLabel = function(theStation) {
-      var cleanString = theStation.replace('@', '');
-      var capitalizedStr = capitalizeStr(cleanString, 8, 9);
-      capitalizedStr = '<strong>' + capitalizedStr + '</strong>';
+      var cleanString = theStation.replace('@', '')
+      var capitalizedStr = capitalizeStr(cleanString, 8, 9)
+      capitalizedStr = '<strong>' + capitalizedStr + '</strong>'
 
-      return capitalizedStr;
-    };
+      return capitalizedStr
+    }
 
 
     var updateView = function(dom, theDate, widgetSelect, title, outputString) {
-      var err;
-      var theLastUpdate = lastUpdated;
+      var err
+      var theLastUpdate = lastUpdated
 
       try {
-        dom.find(widgetSelectTitle).html(widgetSelect);
+        $(document).find(widgetSelectTitle).html(widgetSelect)
```

Comment:
Can't you do $(dom)? I ask because dom is just your widget, whole document is the all widget container, so you might find the wrong thing if you go looking in document

Reply:
Or just use domEl, the thing that Ubersicht actually provides to this function, instead of passing it in again